packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx

```diff
@@ -0,0 +1,142 @@
+        <EntitiesContext.Provider value={contextValue}>
+          <BeanReferenceField name="test" label="Bean reference field label" {...fieldProperties} />
+        </EntitiesContext.Provider>,
+      );
+    });
+    let options = screen.queryAllByRole('option');
+    expect(options).toHaveLength(0);
+    const toggleButton = screen.getByRole('button', { name: 'Menu toggle' });
+    await act(async () => {
+      fireEvent.click(toggleButton);
+    });
+    options = screen.getAllByRole('option');
+    expect(options).toHaveLength(3);
+    const selectedOption = screen.getByRole('option', { selected: true });
+    expect(selectedOption).toHaveTextContent('myDataSource');
+  });
+
+  it('should update model if the other is selected', async () => {
+    await act(async () => {
+      render(
+        <EntitiesContext.Provider value={contextValue}>
+          <BeanReferenceField name="test" label="Bean reference field label" {...fieldProperties} />
+        </EntitiesContext.Provider>,
+      );
+    });
+    const toggleButton = screen.getByRole('button', { name: 'Menu toggle' });
+    await act(async () => {
+      fireEvent.click(toggleButton);
+    });
+    const myBeanOption = screen.getAllByRole('option').filter((option) => option.innerHTML.includes('myBean'));
+    expect(myBeanOption).toHaveLength(1);
+    expect(mockOnChange.mock.calls).toHaveLength(0);
+    act(() => {
+      fireEvent.click(myBeanOption[0]);
+    });
+    expect(mockOnChange.mock.calls).toHaveLength(1);
+    expect(mockOnChange.mock.calls[0][0]).toEqual('#myBean');
+  });
+
+  it("should update model even if there's no bean", async () => {
+    await act(async () => {
+      render(
+        <EntitiesContext.Provider value={contextValue}>
+          <BeanReferenceField name="test" label="Bean reference field label" {...fieldProperties} />
+        </EntitiesContext.Provider>,
+      );
+    });
+    const textbox = screen.getByRole('combobox');
+    expect(mockOnChange.mock.calls).toHaveLength(0);
+    await act(async () => {
+      fireEvent.input(textbox, { target: { value: '#notExistingBean' } });
+    });
+    expect(mockOnChange.mock.calls).toHaveLength(1);
+    expect(mockOnChange.mock.calls[0][0]).toEqual('#notExistingBean');
+  });
+
+  it.skip('should render a modal', async () => {
+    await act(async () => {
+      render(
+        <EntitiesContext.Provider value={contextValue}>
+          <BeanReferenceField name="test" label="Bean reference field label" {...fieldProperties} />
+        </EntitiesContext.Provider>,
+      );
+    });
+    const toggleButton = screen.getByRole('button', { name: 'Menu toggle' });
+    await act(async () => {
+      fireEvent.click(toggleButton);
+    });
+    const createNewOption = screen.getAllByRole('option').filter((option) => option.textContent === 'Create new bean');
+    expect(createNewOption).toHaveLength(1);
+    await act(async () => {
+      //  TypeError: Cannot read properties of undefined (reading 'Provider')
```

Comment:
I tried to test the "create new bean" modal is rendered, but the weird error was thrown from inside the CustomAutoForm. This test is skipped for now.

Reply:
@igarashitm I created this https://github.com/igarashitm/kaoto-next/pull/4 to enable this test.

The **TL;DR** is that `uniforms` fields require a `componentDetector` provider, and to be a descendant of `AutoForm` which holds the schema bridge.

The stack trace you saw is complaining about a nested field without a detector.

Reply:
Thanks! I never thought the outer form is the cause as the error message looked like coming from the inner form, the new bean form in the modal :sweat_smile: